src/unread/unreadModel.js

```diff
@@ -243,8 +244,52 @@ function streamsReducer(
       }
 
       if (action.op === 'remove') {
-        // Zulip doesn't support un-reading a message.  Ignore it.
-        return state;
+        const { message_details } = action;
+        if (message_details === undefined) {
+          logging.warn('Got update_message_flags event without message_details.');
```

Comment:
As is, this is describing a situation that's perfectly fine and normal -- most update_message_flags events won't have message_details. We can make it more specific, to avoid puzzling someone looking at a Sentry event in the future who has an idea how this part of the API works, but has forgotten the details of how we wrote this code:

```suggestion
        if (message_details === undefined) {
          logging.warn('Got update_message_flags/remove/read event without message_details.');
```

(Also to avoid looking wrong for someone reading this code to try to understand it.)